### ISSUE TITLE
Material 3 Date Picker with Disabled Dates: Disabled Dates are Focusable with External Keyboard

### ISSUE THREAD

**Issue by swoods954** [material-components/material-components-android#4169]
**Description:** Disabled dates in the M3 Date Picker component are focusable via external keyboard navigation, however there is no visible focus indicator. Disabled controls should not receive keyboard focus.

Note: This is an accessibility failure of WCAG 1.4.7 Focus Visible (and raises concerns with WCAG 2.1.1 Keyboard as well.

**Expected behavior:** Disabled controls are not focusable via external keyboard navigation. When using <arrow> keys on a month with disabled dates, focus should skip over the disabled dates.

**Steps to Reproduce:**

1. Connect a physical external keyboard to an Android mobile device
2. Open a default Material 3 Date Picker component that contains disabled dates
4. Using the <arrow> keys on the keyboard, navigate to a disabled date
5. Confirm disabled dates can be focused, but focus is “lost” since there is no visible focus indicator for the disabled dates

Note on attachments:
1. Recording of issue from Material 3 Date Picker
2. Bug report from Samsung Galaxy S20 (also reproducible on Google Pixel devices)

https://github.com/material-components/material-components-android/assets/126836844/73388230-4789-419c-aa85-735644e2b9b1

[bugreport-x1quex-TP1A.220624.014-2024-05-01-16-12-22.zip](https://github.com/material-components/material-components-android/files/15280302/bugreport-x1quex-TP1A.220624.014-2024-05-01-16-12-22.zip)

**Android API version:** All

**Material Library version:** Material 3 Libraries

**Device:**  Samsung Galaxy S20 and all Google Pixel
